src/App.tsx

```diff
@@ -2,23 +2,29 @@ import React from 'react';
 import { StatusBar } from 'react-native';
 import { NavigationContainer } from '@react-navigation/native';
 import { createStackNavigator } from '@react-navigation/stack';
+import { useSelector } from 'react-redux';
 import HomeContainer from './containers/Home';
 import LoginContainer from './containers/Login';
 
 const Stack = createStackNavigator();
 
 const App: React.FC = () => {
+  const isLoggedIn = useSelector((state) => (state as any).auth.isLoggedIn);
+
   return (
     <>
       <StatusBar barStyle="dark-content" />
       <NavigationContainer>
         <Stack.Navigator>
-          <Stack.Screen
```

Comment:
Let's remove the use of NavigationContainer and Stack.Screens here if we can. Let's not remove it from the libraries, maybe add it to the Home container so that users of the bootstrap can see an example of it (although it's basically right out of the documentations, but so is everything else). But here we just want to switch between two top level components which you're doing here.

Reply:
What makes most sense to me is to have 1 file define all possible screens in the app, like in `App.tsx` and then all of the screens (like `Home` and `Login`) would just be `Views`, similar to how it's done with `react-router` for web apps. Or am I misunderstanding what you're wanting to do here?

Reply:
Yah I think you got it exactly. But most likely it won't go beyond two, maybe three of these screens so I don't want to figure out anything larger than what you're already doing, but minus the Stack.Screen part.
I'm thinking a full screen loading indicator would also go here in App.tsx. As well as any alerts or other floating dialogs or tooltips we might want will go here eventually.

Reply:
But removing Stack.Screen would just make them all Views on a single screen, right? So you're saying have all the output be on 1 screen?

Wouldn't it make more sense for the bootstrap to have some small navigation to show how to do some navigation while also mimicking a real world app? 

When you download an app that needs authentication, you would first only see a sign up screen. Once you sign up, it should then redirect you to the Home page, save the token in storage, and then every time the app is opened it should go straight to the Home page (or sign in page if the session expires).

Reply:
It would be good to have an example of a navigation screen, so we could put it on the Home screen and if you want to add another "profile" screen or something to show navigation back and forward after the user's logged in, that would be good for people to have in a bootstrap.

But for switching the context between authentication and the rest of the app operation, we'd like to swap out the entire container in App.tsx based on the authentication state. Otherwise, notice how we'll have to add extra code to remove the login screen from the navigation stack after the user logs in (to get rid of that back button). It'll be sharing a navigation context between two containers that have very different flows, which isn't a clean UI hierarchy and will cause complications later. Normally, a pre-authenticated user will need to be shown something different if there's an onboarding flow than a post-authenticated user.

By swapping the components in App.tsx between Home and Login, we'll be sandboxing the navigation stacks so that login can have its own stack for onboarding flows (doesn't need one right now so not necessary to have one), and home will have its own navigation stack for the rest of operations (will most def need one for a real app, but it'd be a nice-to-have as a bootstrap example in this project if you want).

Reply:
That makes a lot of sense. However, the way this works currently, there is no back button between the login screen and the Home Screen, at least not on the iPhone simulator I tested on. I'm pretty sure they're completely different stacks based on the conditional.

It also looks like the react navigation docs recommend this way as well: https://reactnavigation.org/docs/auth-flow/

Could you load this branch up on a simulator and take a look? I get what you're trying to accomplish, but if this PR hasn't already achieved that then I don't know exactly how to so I would need some help on it.